docs/install/authentication-mechanisms-gcp.md

```diff
@@ -131,3 +131,121 @@ gcloud projects add-iam-policy-binding $PROJECT_ID \
 
    Note that `google-cloud-key` and `key.json` are default values expected by
    our control plane.
+
+## Troubleshooting
+
+### Workload Identity
+
+To ensure Workload Identity correctly configured for a Google Cloud Service Account (GSA), 
+and a Kubernetes Service Account (KSA), you need to check:
+
+1. If the Google Cloud Service Account has the desired `iam-policy-binding`.
+   
+   By running the following command: 
+   ```shell
+   gcloud iam service-accounts get-iam-policy gsa-name@$PROJECT_ID.iam.gserviceaccount.com
+   ```
+   if the `iam-policy-binding` is correct, you'll find a `iam-policy-binding` similar to:
+   ```shell
+    bindings:
+    - members:
+      - serviceAccount:$PROJECT_ID.svc.id.goog[ksa-namespace/ksa-name]
+      role: roles/iam.workloadIdentityUser
+   ```
+ 2. If the Kubernetes Service Account has the desired annotation.
+   
+    By running the following command: 
+    ```shell
+    kubectl get serviceaccount ksa-name -o yaml -n ksa-namespace
+    ```
+    if the annotation exists, you'll find an annotation similar to:
+    ```shell
+    metadata:
+      annotations:
+        iam.gke.io/gcp-service-account: gsa-name@$PROJECT_ID.iam.gserviceaccount.com
+    ```
+
+### Kubernetes Secrets
+
+To ensure kubernetes Secret correctly configured in a namespace, you need to check if a Kubernetes Secret 
+named `google-cloud-key` is in desired namespace by running the following command: 
+
+```shell
+kubectl get secret -n secret-namespace
+```
+
+### Common Issues
+
+* ***Resources are not READY***
+
+   If a resource instance is not READY due to authentication configuration problem, 
+   you are likely to get `PermissionDenied` error for `User not authorized to perform this action`
+   (for Workload Identity, the error might look like `IAM returned 403 Forbidden: The caller does not have permission`).
+   
+   This error indicates that the Control Plane may not configure authentication properly. You can find it in a resource instance by:
+   
+   ```shell
+   kubectl describe resource-name resource-instance-name -n resource-instance-namespace
+   ```
+   
+   Here is an example checking a CloudAuditlogsSource `test` in namespace `default`:
+      
+   ```shell
+   kubectl describe cloudauditlogssource test -n default
+   ```
+   
```

Comment:
Include the relevant portions of the command's output for both secret and WI problems.

Essentially, something like the following (each should probably be its own section). I want people to be able to easily search for the failure reason string (e.g. Google search `'User not authorized to perform this action'` and find this page):


If this CloudAuditLogsSource failed due to a missing Secret, then it would look like:

```
Condition
```

If this CloudAuditLogsSource failed due to a malformed Secret, then it would look like:

```
Condition
```

If this CloudAuditLogsSource failed due to a valid Secret but the GSA has insufficient permissions, then it would look like:

```
Type: Ready
Value: False
Reason: PermissionDenied
Message: User not authorized to...
```

If this CloudAuditLogsSource failed due to the IAM Secret having been revoked, then it would look like:

```
Condition
```

If this CloudAuditLogsSource failed due to WI being setup incorrectly, then it would look like:

```
Condition
```

If this CloudAuditLogsSource failed due to WI working, but lacking sufficient permissions, then it would look like:


```
Condition
```

Reply:
added